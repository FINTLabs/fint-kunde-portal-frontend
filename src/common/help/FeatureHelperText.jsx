import PropTypes from "prop-types";
import React, { Component } from "react";
import { withStyles, Paper } from "@material-ui/core";
import classNames from "classnames";
import Box from "@material-ui/core/Box";

const styles = theme => ({
  helpText: {
    marginBottom: theme.spacing(2),
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
    paddingTop: theme.spacing(),
    paddingBottom: theme.spacing(),
    backgroundColor: theme.palette.grey[200],
    fontFamily: theme.typography.body1.fontFamily,
    fontSize: theme.typography.body1.fontSize,
    fontWeight: theme.typography.body1.fontWeight
  },
  closeButton: {
    float: "right",
    cursor: "pointer",
    color: theme.palette.grey[500],
    "&:hover": {
      color: theme.palette.grey[700]
    }
  },
  closeButtonContainer: {
    height: theme.spacing(1.5)
  },
  hide: {
    display: "none"
  }
});

class FeatureHelperText extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      show: props.show
    };
  }

  close = () => {
    this.setState({ show: false });
  };

  render() {
    const { classes } = this.props;
    const { show } = this.state;
    const classHidden = classNames({
      [classes.hide]: !show
    });
    return (
      <div className={classHidden}>
        <Box className={classes.helpText} borderRadius={5} mt={4}>{this.props.children}</Box>
      </div>
    );
    /*
    return (
      <div className={classHidden}>
        <Paper className={classes.helpText}>
          <div className={classes.closeButtonContainer}>
            <CloseIcon className={classes.closeButton} onClick={() => this.close()}/>
          </div>
          {this.props.children}
        </Paper>
      </div>
    );
    */
  }
}

FeatureHelperText.defaultProps = {
  show: true
};

FeatureHelperText.propTypes = {
  children: PropTypes.any.isRequired,
  classes: PropTypes.object.isRequired,
  show: PropTypes.bool.isRequired
};
export default withStyles(styles)(FeatureHelperText);
