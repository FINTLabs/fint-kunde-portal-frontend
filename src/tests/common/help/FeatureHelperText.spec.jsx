import React from "react";
import { create } from "react-test-renderer";
import FeatureHelperText from "../../../common/help/FeatureHelperText";
describe("Feature component", () => {
  const component = create(
    <FeatureHelperText show={true}>Test</FeatureHelperText>
  );
  it("should match snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
