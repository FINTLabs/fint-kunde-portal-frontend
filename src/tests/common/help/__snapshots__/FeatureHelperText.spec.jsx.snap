// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature component should match snapshot 1`] = `
<div
  className=""
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 FeatureHelperText-helpText-1 MuiPaper-rounded"
  >
    Test
  </div>
</div>
`;
